test/functional/api/buffer_updates_spec.lua

```diff
@@ -317,6 +319,8 @@ describe('API: buffer events:', function()
   end)
 
   it('when lines are filtered', function()
+    skip(funcs.executable('cat') == 0, 'missing "cat" command')
```

Comment:
`cat` is not a reasonable thing to be missing. Silently skipping tests when requirements are missing, is something we should avoid.

Reply:
Then I suppose the correct action would be to ensure neovim uses the cat that is shipped during tests?

Reply:
Or check that `cat` behaves as expected and throw an error if not. (I.e. "feature detection" rather than "identity check")

Are we encountering cases where people have a misbehaving `cat` ?

Reply:
We're having cases where these tests are failing locally on windows because cat is missing. On CI they're passing because presumably cat is already avilable by default.

Reply:
weird, I thought the tests already use the  `cat` that comes with Nvim. Or at least they should be able to find it as a last resort. 

so yes, the tests should try to use Nvim's `cat` fallback... and show an error if a reasonable `cat` is not found

Reply:
> weird, I thought the tests already use the cat that comes with Nvim. 

They did before this PR: https://github.com/neovim/neovim/pull/14402

Reply:
That PR is a bit too strict imo. Shipping `cat.exe` should be fine in terms of security.